src/profile/helpers/getEmailsFromNode.ts

```diff
@@ -0,0 +1,26 @@
+import {
+  MyProfileQuery,
+  MyProfileQuery_myProfile_emails_edges_node as Email,
+} from '../../graphql/generatedTypes';
+
+const getEmailsFromNode = (data?: MyProfileQuery) => {
+  if (data?.myProfile?.emails?.edges) {
+    return data.myProfile.emails.edges
+      .map(edge => {
+        if (edge?.node) {
+          return {
+            email: edge.node.email,
+            id: edge.node.id,
+            primary: edge.node.primary,
+            emailType: edge.node.emailType,
+          } as Email;
+        }
+        return false;
+      })
+      .filter((email): email is Email => Boolean(email))
+      .filter((email: Email) => !email.primary);
```

Comment:
I think that this is sort of hard to read 🤔Here's a quick suggestion on how you can make it a bit simpler.
```js
edge
  .filter(edge => edge.node && edge.node.primary)
  .map(edge => ({
    email: edge.node.email,
    id: edge.node.id,
    primary: edge.node.primary,
    emailType: edge.node.emailType,
  }))
```

You could do something like
```js
const edges = data?.myProfile?.emails?.edges || []; // don't know the correct TS pattern
```
to avoid multiple returns as well.

Reply:
Thanks for suggestion! I was struggling to find a solution that would preserve required types.

Reply:
Oh, ok. I imagined there was something like that going on! I have no idea if types are retained with that solution.

Reply:
Seems to be working just fine. I think my mistake was to add `if` check inside map, that resulted typing to be `(Email | undefined)[]`.